test/jdk/java/util/HashMap/WhiteBoxResizeTest.java

```diff
@@ -336,7 +339,25 @@ List<Object[]> genRequestedSizeCases(int size, int cap) {
         return Arrays.asList(
                 rsc("rshm", size, cap, () -> HashMap.newHashMap(size)),
                 rsc("rslm", size, cap, () -> LinkedHashMap.newLinkedHashMap(size)),
-                rsc("rswm", size, cap, () -> WeakHashMap.newWeakHashMap(size))
+                rsc("rswm", size, cap, () -> WeakHashMap.newWeakHashMap(size)),
+                rsc("rshs", size, cap, () -> {
+                    try {
+                        Field mapField = HashSet.class.getDeclaredField("map");
+                        HashSet<String> set = HashSet.newHashSet(size);
+                        return (Map<String, String>) mapField.get(set);
+                    } catch (Exception e) {
+                        throw new RuntimeException(e);
+                    }
+                }),
+                rsc("rsls", size, cap, () -> {
+                    try {
+                        Field mapField = HashSet.class.getDeclaredField("map");
```

Comment:
For clarity, consider using var handle/method handle and keep the reflection code in the `WhiteBoxResizeTest` constructor.

Reply:
@liach I refactored the tests. please have a look.